src/agent/tests/ChatTest.php

```diff
@@ -0,0 +1,173 @@
+use PHPUnit\Framework\TestCase;
+use Symfony\AI\Agent\AgentInterface;
+use Symfony\AI\Agent\Chat;
+use Symfony\AI\Agent\Chat\MessageStoreInterface;
+use Symfony\AI\Platform\Message\AssistantMessage;
+use Symfony\AI\Platform\Message\Message;
+use Symfony\AI\Platform\Message\MessageBag;
+use Symfony\AI\Platform\Message\MessageBagInterface;
+use Symfony\AI\Platform\Result\TextResult;
+
+#[CoversClass(Chat::class)]
+#[UsesClass(Message::class)]
+#[UsesClass(MessageBag::class)]
+#[UsesClass(TextResult::class)]
+#[Small]
+final class ChatTest extends TestCase
+{
+    private AgentInterface&\PHPUnit\Framework\MockObject\MockObject $agent;
+    private MessageStoreInterface&\PHPUnit\Framework\MockObject\MockObject $store;
+    private Chat $chat;
+
+    protected function setUp(): void
+    {
+        $this->agent = $this->createMock(AgentInterface::class);
+        $this->store = $this->createMock(MessageStoreInterface::class);
+        $this->chat = new Chat($this->agent, $this->store);
+    }
+
+    #[Test]
+    public function itInitiatesChatByClearingAndSavingMessages(): void
+    {
+        $messages = $this->createMock(MessageBagInterface::class);
+
+        $this->store->expects(self::once())
+            ->method('clear');
+
+        $this->store->expects(self::once())
+            ->method('save')
+            ->with($messages);
+
+        $this->chat->initiate($messages);
+    }
+
+    #[Test]
+    public function itSubmitsUserMessageAndReturnsAssistantMessage(): void
+    {
+        $userMessage = Message::ofUser('Hello, how are you?');
+        $existingMessages = new MessageBag();
+        $assistantContent = 'I am doing well, thank you!';
+
+        $textResult = new TextResult($assistantContent);
+
+        $this->store->expects(self::once())
+            ->method('load')
+            ->willReturn($existingMessages);
+
+        $this->agent->expects(self::once())
+            ->method('call')
+            ->with(self::callback(function (MessageBagInterface $messages) use ($userMessage) {
+                $messagesArray = $messages->getMessages();
+
+                return end($messagesArray) === $userMessage;
+            }))
+            ->willReturn($textResult);
+
+        $this->store->expects(self::once())
+            ->method('save')
+            ->with(self::callback(function (MessageBagInterface $messages) use ($userMessage, $assistantContent) {
+                $messagesArray = $messages->getMessages();
+                $lastTwo = \array_slice($messagesArray, -2);
+
+                return 2 === \count($lastTwo)
+                    && $lastTwo[0] === $userMessage
+                    && $lastTwo[1] instanceof AssistantMessage
+                    && $lastTwo[1]->content === $assistantContent;
+            }));
+
+        $result = $this->chat->submit($userMessage);
+
+        self::assertInstanceOf(AssistantMessage::class, $result);
+        self::assertSame($assistantContent, $result->content);
```

Comment:
will fail after rebase due to #202 
```suggestion
        $this->assertInstanceOf(AssistantMessage::class, $result);
        $this->assertSame($assistantContent, $result->content);
```

Reply:
Will rebase an execute cs fixer